From 16aa4c9a1fbe763c147a964cdc1f5be8ed98ed13 Mon Sep 17 00:00:00 2001
From: Zhang Yi <yi.zhang@huawei.com>
Date: Wed, 30 Jun 2021 16:36:38 +0800
Subject: [PATCH] jbd2: export jbd2_journal_[un]register_shrinker()
Git-commit: 16aa4c9a1fbe763c147a964cdc1f5be8ed98ed13
Patch-mainline: v5.14-rc1
References: bsn#12


Export jbd2_journal_[un]register_shrinker() to fix this error when
ext4 is built as a module:

  ERROR: modpost: "jbd2_journal_unregister_shrinker" undefined!
  ERROR: modpost: "jbd2_journal_register_shrinker" undefined!

Fixes: 4ba3fcdde7e3 ("jbd2,ext4: add a shrinker to release checkpointed buffers")
Signed-off-by: Zhang Yi <yi.zhang@huawei.com>
Reviewed-by: Jan Kara <jack@suse.cz>
Link: https://lore.kernel.org/r/20210630083638.140218-1-yi.zhang@huawei.com
Signed-off-by: Theodore Ts'o <tytso@mit.edu>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 fs/jbd2/journal.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/jbd2/journal.c
+++ b/fs/jbd2/journal.c
@@ -2028,6 +2028,7 @@ int jbd2_journal_register_shrinker(journ
 
 	return 0;
 }
+EXPORT_SYMBOL(jbd2_journal_register_shrinker);
 
 /**
  * jbd2_journal_unregister_shrinker()
@@ -2040,6 +2041,7 @@ void jbd2_journal_unregister_shrinker(jo
 	percpu_counter_destroy(&journal->j_jh_shrink_count);
 	unregister_shrinker(&journal->j_shrinker);
 }
+EXPORT_SYMBOL(jbd2_journal_unregister_shrinker);
 
 /**
  * jbd2_journal_destroy() - Release a journal_t structure.
